design/src/rfcs/rfc0004_retry_behavior.md

```diff
@@ -0,0 +1,147 @@
+RFC: Retry Behavior
+============================
+
+> Status: RFC
+
+For a summarized list of proposed changes, see the [Changes Checklist](#changes-checklist) section.
+
+It is not currently possible for users of the SDK to configure a client's retry behavior. This RFC establishes a method for users to set the number of retries to attempt when calling a service and would allow users to disable retries entirely.
+
+Terminology
+-----------
+
+- **Shared Config**: An `aws_config::Config` struct that is responsible for storing shared configuration data that is used across all services. This is not generated and lives in the `aws-config` crate.
+- **Service-specific Config**: A code-generated `Config` that has methods for setting service-specific configuration. Each `Config` is defined in the `config` module of its parent service. For example, the S3-specific config struct is `use`able from `aws_sdk_s3::config::Config` and re-exported as `aws_sdk_s3::Config`.
+
+Retry config
+--------------------
+
+Users can set the number of retries in several ways:
+- By setting the `AWS_MAX_ATTEMPTS` environment variable
+- By calling the `aws_config::ConfigLoader::retry_config(..)` method
+- By calling the retry_config method on a service-specific config
+
+The above list is in order of increasing precedence e.g. setting retry attempts with the `retry_config` method will override a value set by `AWS_MAX_ATTEMPTS`.
+
+## Setting an environment variable
```

Comment:
Did you intend to make these the same heading level as the Retry Config section?

Reply:
Good catch, thanks. I'm not used to the ----- and ===== style headings